runtime/autoload/man.vim

```diff
@@ -1,6 +1,15 @@
-let s:man_tag_depth = 0
+" Ensure Vim is not recursively invoked (man-db does this)
+" by removing MANPAGER from the environment
+" More info here http://comments.gmane.org/gmane.editors.vim.devel/29085
+if &shell =~# 'fish$'
+  let s:man_cmd = 'man -P cat ^/dev/null '
+else
+  let s:man_cmd = 'man -P cat 2>/dev/null '
+endif
+
 let s:man_sect_arg = ''
 let s:man_find_arg = '-w'
+let s:tag_stack = []
 
 try
   if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~? '^5'
```

Comment:
@justinmk I'm removing this because its a fix for only Solaris/SunOS and we do not support them.